### PR TITLE
Update ECIES implementation for crypto 3.2

### PR DESCRIPTION
**Pull request by romanstrobl**
This pull request introduces updated ECIES implementation for crypto 3.2.

### CODE REVIEW

File: powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/steps/AbstractBaseStep.java [hunk 1]
```diff
@@ -205,26 +214,51 @@ public void addEncryptedRequest(StepContext<M, R> stepContext, String applicatio
...
+        try {
+            SimpleSecurityContext securityContext = (SimpleSecurityContext) stepContext.getSecurityContext();
+            EciesEncryptedResponse encryptedResponse = stepContext.getResponseContext().getResponseBodyObject();
+            final byte[] transportMasterKeyBytes = Base64.getDecoder().decode(stepContext.getModel().getResultStatus().getTransportMasterKey());
+            final byte[] nonceBytes = encryptedResponse.getNonce() != null ? Base64.getDecoder().decode(encryptedResponse.getNonce()) : null;
+            EciesParameters eciesParameters = EciesParameters.builder()
+                    .nonce(nonceBytes)
+                    .timestamp(encryptedResponse.getTimestamp())
+                    .build();
+            String applicationSecret = stepContext.getModel().toMap().get("APPLICATION_SECRET").toString();
```
Comment (banterCZ) [inline powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/steps/AbstractBaseStep.java:231]: Is that NPE proof?

File: powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/steps/AbstractBaseStep.java [hunk 2]
```diff
@@ -205,26 +214,51 @@ public void addEncryptedRequest(StepContext<M, R> stepContext, String applicatio
...
+                    getStep().id() + "-response-decrypt",
+                    "Decrypted Response",
+                    "Following data were decrypted",
+                    "OK",
+                    responsePayload
+            );
+
+            return responsePayload;
+        } catch (Exception ex) {
+            return null;
```
Comment (banterCZ) [inline powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/steps/AbstractBaseStep.java:260]: May we log it at least for `TRACE`?

**Review by banterCZ — APPROVED** [review]
Looks good.

File: powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/util/EncryptionUtil.java [hunk 3]
```diff
@@ -54,4 +84,25 @@ public static void processEncryptedResponse(StepContext<?, EciesEncryptedRespons
...
+            if (eciesScope == EciesScope.ACTIVATION_SCOPE) {
+                return ByteUtils.concatStrings(version.value(), applicationKey, activationId);
+            } else {
+                return ByteUtils.concatStrings(version.value(), applicationKey);
+            }
+        } else {
+            return null;
+        }
+    }
+
```
Comment (hvge) [inline powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/util/EncryptionUtil.java:107]: This should be a part of `powerauth-crypto` library. We have the same code duplicated in the server's implelementation.

Comment (romanstrobl) [reply powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/util/EncryptionUtil.java:107]: Ah, yes, I wanted to move this class.

**Pull request merged**